lib/datastruct/kvldskey.c

```diff
@@ -15,12 +15,15 @@ kvldskey_create(const uint8_t * buf, size_t len)
 {
```

Comment:
Leaving the API as `size_t` is good.  This way future bugs are more likely to be caught via the assertion rather than hidden by silent integer overflow.